### DISCUSSION TITLE
BarChart - Thresholds do not updates when data is updated

### DISCUSSION THREAD

**Discussion by melquelima** [grafana/grafana#57790] [Alerting]
Hey all!

i am using BarChart and Thresholds to color the bars on a graphic, i noticed that when the data is updated automatically by the page it does not draw again the rules for Thresholds colors it keeps the color, it changes onl when i refresh the page, some idea how do i solve it?

**Comment by leeoniya** [comment]
can you open a new issue with your grafana version, panel data, and config?

are you using `Color by field` + field threshold overrides? are you using `Color scheme: From thresholds (by value)`? are your thresholds dynamic & coming from data?

i do see that we have some issues with coloring from thresholds or value mappings when these are only set in the general panel options instead of explicitly set on a field with `Color by field` configured.

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
